dspace-api/src/main/java/org/dspace/versioning/dao/impl/VersionDAOImpl.java

```diff
@@ -26,14 +30,36 @@
  * @author Mark Diggory (markd at atmire dot com)
  * @author Ben Bosman (ben at atmire dot com)
  * @author kevinvandevelde at atmire.com
+ * @author Pascal-Nicolas Becker (dspace at pascal dash becker dot de)
  */
 public class VersionDAOImpl extends AbstractHibernateDAO<Version> implements VersionDAO
 {
-
     @Override
     public Version findByItem(Context context, Item item) throws SQLException {
         Criteria criteria = createCriteria(context, Version.class);
         criteria.add(Restrictions.eq("item", item));
         return singleResult(criteria);
     }
+
+    @Override
+    public int getNextVersionNumber(Context c, VersionHistory vh) throws SQLException {
+        Query q = this.createQuery(c, 
+                "SELECT (COALESCE(MAX(versionNumber), 0) + 1) "
+                        + "FROM Version WHERE versionHistory.id = :historyId");
+        q.setParameter("historyId", vh.getId());
+
+        int next = (Integer) q.uniqueResult();
+        return next;
```

Comment:
I would use the utility method from AbstractHibernateDAO: return count(q);. This will suppress the casting warning.